src/multiSceneMovie.coffee

```diff
@@ -28,23 +29,47 @@ class MultiSceneMovie
   composeScenes = (rootElement, movie, movieScene, movieLoadScene, movieFinishScene, contentScene) ->
     ee = new EventEmitter(rootElement)
 
-    bindTrackingEvent = (event, url, type)->
-      switch event
-        when "creative_view"
-          ee.listen("movie:screen:appeared", -> new Beacon(url, type))
-        when "start"
-          ee.listen("movie:started", -> new Beacon(url, type))
-        when "first_quartile"
-          ee.listen("movie:played:firstQuartile", -> new Beacon(url, type))
-        when "mid_point"
-          ee.listen("movie:played:half", -> new Beacon(url, type))
-        when "third_quartile"
-          ee.listen("movie:played:thirdQuartile", -> new Beacon(url, type))
-        when "complete"
-          ee.listen("movie:finished", -> new Beacon(url, type))
-
-    for tracking in rootElement.getElementsByClassName("tracking")
-      bindTrackingEvent(tracking.dataset.event, tracking.dataset.url, tracking.dataset.type)
+    composeVastTrackingEvents = (eventEmitter) ->
+      bindTrackingEvent = (eventEmitter, tracking) ->
+        return unless eventEmitter
+        eventName = undefined
+
+        switch tracking.event
+          when "creative_view"
```

Comment:
It might be a little more maintainable if you set up an associative array mapping event names?